src/main/java/org/opensearch/ad/AnomalyDetectorExtension.java

```diff
@@ -111,6 +112,12 @@ public OpenSearchClient getClient() {
         return client;
     }
 
+    // TODO: After AnomalyDetectorJob.java is uncommented, fetch jobIndex from AnomalyDetectorJob
+    @Override
+    public JobDetails getJobDetails() {
+        return new JobDetails(AnomalyDetectorPlugin.AD_JOB_TYPE, ".opendistro-anomaly-detector-jobs");
```

Comment:
We should use Job index constant here `ANOMALY_DETECTOR_JOB_INDEX`. 
Also, how does `getJobDetails` is equivalent to which JobScheduler extension point here?
For ADplugin. we have `getJobIndex()`

Reply:
@owaiskazi19 `JobDetails` is equivalent to both `getJobIndex()` and `getJobType()`. Both these strings are utilized by Job Scheduler to configure it's behavior for `onIndexModule()` and to also configure the `indexToJobProviders` which the `JobSweeper` uses to schedule jobs. I also agree that we should use the `ANOMALY_DETECTOR_JOB_INDEX` constant, @vibrantvarun it should be fine to uncomment this

Reply:
I want to understand more on why are we coupling up `getJobIndex()` and `getJobType()` together in `jobDetails`?Shouldn't it follow the old design or otherwise we have to change these extension points in all the plugins/extensions using JS?

Reply:
`getJobIndex` and `getJobType` are currently used by Job Scheduler to retrieve implementations of the `JobSchedulerExtension` interface [here](https://github.com/opensearch-project/job-scheduler/blob/7727871496c8c9aceebd415bf8169dea6284c938/src/main/java/org/opensearch/jobscheduler/JobSchedulerPlugin.java#L125). Simply put, these two extension points just provide strings to Job Scheduler.

We're coupling the Job Index and Job Type strings together in one request, rather than separate requests since this is the only initial information we need from extensions in order to configure both the `indicesToListen` set and `indexToJobProviders` hashmap. Separating these into two requests would be unnecessary.  

This wouldn't have any affect on how plugins are using JS, since the plugin architecture for JS remains untouched within `PluginsService`'s `loadExtensionsForPlugin` workflow. As for extensions, the only difference at this point is how we are providing these strings to Job Scheduler. How Job Scheduler uses these strings will remain unchanged.

Reply:
I understand separating them would be unnecessary but that's how the old design is. We don't to bring new extension points with extensibility at this moment before covering the old ones.

Reply:
Okay @owaiskazi19 . 

Following your comment

1) I will be deleting the Job Details object and then decouple both the parameters
2) I will also be making 2 transportService calls now to fetch both the parameters

Reply:
Hi @joshpalis @owaiskazi19 the job index cannot be pulled from AnomalyDetectorJob.java as of now because it still has circular dependency from JS on some other methods written in the file.  If I uncomment only the required code like the Job Index initialization then I am facing an error of jacoco test coverage report. Therefore, for now I am hardcoding the value of job index.

Reply:
We can fix the jacoco test coverage, right? The way you fixed it for ModelTests.

Reply:
Yes I did uncomment the code, but there are some dependencies which are imported from JobScheduler like Schedule.java etc. These dependencies cannot be resolved until they are imported from JS native plugin.

Reply:
Makes sense. Let's create a constant in this file then

Reply:
ACK